components/views/dropdown-menu-view.js

```diff
@@ -0,0 +1,262 @@
+define([
+        'backbone',
+        'jquery',
+        'underscore',
+        'text!../templates/dropdown.underscore'
+    ],
+    function( Backbone, $, _, DropdownTpl ) {
+        'use strict';
+
+        /**
+         *  Dropdown Menu View which is fully accessible
+         *
+         *  Initialize the view by passing in the following attributes
+         *  className: 'space separated string of classes for element',
+         *  model: with the following attributes (example values added)
+         *      main: {
+         *          text: 'username',
+         *          url: 'dashboard'
+         *      },
+         *      button_label: 'User options dropdown',
+         *      items: [
+         *          {
+         *              text: 'Dashboard',
+         *              url: 'dashboard'
+         *          }, {
+         *              text: 'Account',
+         *              url: 'account_settings'
+         *          }, {
+         *              text: 'Profile',
+         *              url: 'learner_profile'
+         *          }, {
+         *              text: 'Sign Out',
+         *              url: 'logout'
+         *          }
+         *      ]
+         *  parent: 'selecter for parent element that will be replaced with dropdown menu'
+         *
+         */
+        return Backbone.View.extend({
+            tpl: _.template( DropdownTpl ),
+
+            events: {
+                'click .js-dropdown-button': 'clickOpenDropdown',
+                'click a': 'analyticsLinkClick',
+                'keydown': 'viewKeypress'
+            },
+
+            menu: '.dropdown-menu',
+
+            key: {
+                tab: 9,
+                enter: 13,
+                esc: 27,
+                up: 38,
+                down: 40
+            },
+
+            initialize: function( options ) {
+                this.$parent = $(options.parent);
+                this.render();
+            },
+
+            className: function() {
+                return this.options.className;
+            },
+
+            render: function() {
+                this.$el.html( this.tpl( this.model.toJSON() ) );
+                this.$parent.replaceWith( this.$el );
+                this.postRender();
+
+                return this;
+            },
+
+            postRender: function() {
+                this.$menu = this.$el.find('.dropdown-menu');
+                this.$page = $(document);
+                this.$dropdownButton = $('.js-dropdown-button');
+                this.listenForPageKeypress();
+            },
+
+            analyticsLinkClick: function( event ) {
+                /**
```

Comment:
I don't understand this comment. Are you suggesting that folks should override this class to add the tracking? They won't be modifying this code directly, right?